### PR TITLE
In many cases user avatar link should be an absolute URL with http host

### PR DESCRIPTION
**Pull request by wxiaoguang**
In many cases user avatar link should be an absolute URL with http host. eg: for API, for webhook.

The last PR (#17123) about user avatar refactoring make `AvatarLink` always returns relative URL, it is incorrect.

This PR makes `AvatarLink` return the absolute URL as before.

### CODE REVIEW

**Review by jolheiser — APPROVED** [review]
LGTM assuming tests pass

File: models/user_avatar.go [hunk 1]
```diff
@@ -91,9 +92,13 @@ func (u *User) AvatarLinkWithSize(size int) string {
 	return avatars.GenerateEmailAvatarFastLink(u.AvatarEmail, size)
 }
 
-// AvatarLink returns a avatar link with default size
+// AvatarLink returns the full avatar link with http host
 func (u *User) AvatarLink() string {
-	return u.AvatarLinkWithSize(0)
+	link := u.AvatarLinkWithSize(0)
+	if !strings.HasPrefix(link, "//") && !strings.Contains(link, "://") {
```
Comment (delvh) [inline models/user_avatar.go:98]: In which case would the link start with `//`? And why do you even test for it?
The second condition makes sense, but why the first one?

File: integrations/user_avatar_test.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func TestUserAvatar(t *testing.T) {
 
 		user2 = db.AssertExistsAndLoadBean(t, &models.User{ID: 2}).(*models.User) // owner of the repo3, is an org
 
-		req = NewRequest(t, "GET", user2.AvatarLink())
+		req = NewRequest(t, "GET", user2.AvatarLinkWithSize(0))
```
Comment (delvh) [inline integrations/user_avatar_test.go:76]: Why did you change that?

File: models/user_avatar.go (hunk 1 above)
Comment (wxiaoguang) [reply models/user_avatar.go:98]: old logic. google: `url double prefix slash`

File: integrations/user_avatar_test.go (hunk 2 above)
Comment (wxiaoguang) [reply integrations/user_avatar_test.go:76]: in test code, the request/server is mocked. full url doesn't make sense.

**Review by lunny — APPROVED** [review]
